worker/sshserver/server.go

```diff
@@ -21,17 +20,29 @@ import (
 
 // ServerWorkerConfig holds the configuration required by the server worker.
 type ServerWorkerConfig struct {
+	// Logger holds the logger for the server.
 	Logger Logger
+
 	// Listener holds a listener to provide the server. Should you wish to run
 	// the server on a pre-existing listener, you can provide it here. Otherwise,
 	// leave this value nil and a listener will be spawned.
 	Listener net.Listener
+
+	// JumpHostKey holds the host key for the jump server.
+	JumpHostKey string
```

Comment:
i think the type should be `ssh.Signer` instead of `string`, that way you catch any errors parsing the specified key outside of the constructor.

Reply:
We could parse it higher up, but it's consumed by the server. So I thought in the server makes sense to me. I would've done parsing even higher up within the state method, but that didn't feel right. I don't mind doing it there but I thought this was OK.

Reply:
there's validation higher up in the CLI layer already

Reply:
would prefer this to be a signer, but ymmv